modules/prebidServerBidAdapter/index.js

```diff
@@ -569,6 +593,8 @@ const OPEN_RTB_PROTOCOL = {
         utils.deepSetValue(imp, 'ext.prebid.storedauctionresponse.id', storedAuctionResponseBid.storedAuctionResponse.toString());
       }
 
+      _appendFloor(firstBidRequest.bids[0], imp);
```

Comment:
I can't figure out what should go here in the first argument

Reply:
Yeah, we'll need some logic to find the right bid to use.

Need's the context of the adUnit, so we'll want to use `adUnit.code`, and find a bid in `firstBidRequest.bids` which matches it.

Something like:

```js
let matchedBid = find(firstBidRequest.bids, bid => bid.adUnitCode === adUnit.code);
if (matchedBid) {
   _appendFloor(matchedBid, imp);
}
```

An issue even this may have:

- `firstBidRequest` may be from a bidder who is not included in all adUnits, so we may miss out on setting floor for some imps.

Which means we may need / want to do a join on ALL bids not just the ones in `firstBidRequest`.

I would have to take a deeper look, but could come up with something.

@bszekely any input on using PBJS floors module to set `imp.bidFloor`?

I know PBS will eventually have some floor module support, but may be some time from now.

Reply:
thanks so much for taking a look! @bszekely1 and I were talking about this in the prebid server meeting recently and it seems this is quite a valid use case. @bretg recently filed https://github.com/prebid/Prebid.js/issues/5720, which this partially solves, partly based on that discussion. Also all prebid server adapters are being asked to honor this parameter when it is available (imp[].bidfloor and imp[].bidlfoorcur) instead of reading from adapterOptions, but there is no way to currently write to it from js.